pkg/controller/injector/k8s_binding_injector.go

```diff
@@ -64,15 +63,10 @@ func (b *k8sBindingInjector) Inject(binding *servicecatalog.Binding, cred *broke
 		Data: make(map[string][]byte),
 	}
 
-	// For each item in the cred just serialize its value into JSON and
-	// save it in the secret
+	// For each item in the cred, just convert it to the bytes for a string
+	// representation of the value
 	for k, v := range *cred {
-		data, err := json.Marshal(v)
-		if err != nil {
-			return fmt.Errorf("Unable to marshal credential value %q: %s",
-				k, err)
-		}
-		secret.Data[k] = data
+		secret.Data[k] = []byte(fmt.Sprintf("%v", v))
```

Comment:
using %v scares me a little. If we're going to do this can we add, as part of this PR, a set of new testcases that verifies that all of the different types of JSON objects that might appear as values are round-tripped properly?